uast/uast-kotlin-fir/src/org/jetbrains/uast/kotlin/internal/firKotlinInternalUastUtils.kt

```diff
@@ -19,6 +25,14 @@ val firKotlinUastPlugin: FirKotlinUastLanguagePlugin by lz {
         ?: FirKotlinUastLanguagePlugin()
 }
 
+internal fun KtType.toPsiClass(context: KtElement): PsiClass? {
```

Comment:
`KtType` should never be passed as receiver and leaked outside analyse call. Please, consider rewriting the code as the following:

```
internal fun KtAnalysisSession.toPsiClass(type: KtType, context: KtElement): PsiClass? {
      return asPsiType(context, TypeMappingMode.DEFAULT_UAST)?.let {
            PsiTypesUtil.getPsiClass(it)
    }
}

```
and always call `toPsiClass` insdie the `analyseForUast` lambda

Reply:
Good to learn that. Rewritten as per the suggestion, thanks.